.github/workflows/weekly.yml

```diff
@@ -35,3 +35,44 @@ jobs:
                 "beta": "true",
               },
             })
+
+      - name: Generate version.txt
+        run: ./gradlew versionFile
+
+      - name: Update changelog_master.xml
+        run: |
+          VERSION=$(cat version.txt | sed 's/-.*//')
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+          # Update the version in changelog_master.xml (the command is only working on GNU sed)
+          sed -i -E '/<release [^>]*version="[^"]+"/ s/(version=")[^ -]+/\1'"$VERSION"'/' app/src/main/res/xml/changelog_master.xml
+
+      - name: Create changelog update branch
+        env:
+          GITHUB_TOKEN: ${{ secrets.TAG_PUSH_TOKEN }}
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+          git fetch origin
+          git checkout -B weekly_changelog_bump
+          git add app/src/main/res/xml/changelog_master.xml
+          git commit -m "Bump changelog_master.xml for weekly release"
+          git push -f origin weekly_changelog_bump
+
+      - name: Create change log update PR
+        uses: actions/github-script@v7
+        env:
+          VERSION: ${{ env.VERSION }}
+        with:
+          script: |
+            const version = process.env.VERSION;
+            github.rest.pulls.create({
+              title: `Bump changelog_master.xml for weekly release ${version}`,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              head: 'weekly_changelog_bump',
+              base: 'main',
+              body: [
+                'Automated PR to update changelog_master.xml for weekly release.',
+                'Double check if we didn't forget to update the changelog changes.',
+              ].join('\n'),
```

Comment:
Open for any suggestion on the content of the PR.

Reply:
> after each weekly workflow run 

With that timing, the text to check for if we forgot to mention anything doesn't make a lot of sense. We're already too late.